packages/iov-bns/src/bnsconnection.ts

```diff
@@ -435,6 +436,22 @@ export class BnsConnection implements BcpAtomicSwapConnection {
     return this.changeTx([bnsNonceTag(addr)]);
   }
 
+  public async getHeader(height: number): Promise<Header> {
+    const { blockMetas } = await this.tmClient.blockchain(height, height);
+    if (blockMetas.length < 1) {
+      throw new Error(`Header ${height} doesn't exist yet`);
+    }
+    const { header } = blockMetas[0];
+    if (header.height !== height) {
+      throw new Error(`Requested header ${height} but got ${header.height}`);
+    }
+    return header;
+  }
+
+  public watchHeaders(): Stream<Header> {
```

Comment:
This obsoletes `public changeBlock(): Stream<number>` once it goes into BcpConnection, right?

Reply:
The other just returns a stream of numbers, this is a bit more complex.
But yes, when there is a clear interface, we can replace that in BcpConnection